src/Scene.h

```diff
@@ -67,7 +73,17 @@ class CScene
 	 */
 	bool intersect(Ray& ray) const
 	{
-		// --- PUT YOUR CODE HERE ---
+		// --- PUT YOUR CODE HERE --
+		/*bool check = false;
+		for (auto& current : m_vpPrims)
+			check |= current->intersect(ray);
+		return check;
+		*/
+		for (auto i = m_vpPrims.begin(); i != m_vpPrims.end(); i++) {
+			if ((*i)->intersect(ray)) {
+				return true;
```

Comment:
you exit the method once the _first_ intersection was found - but you need to find the _nearest_ one!